R/commonQualityControl.R

```diff
@@ -444,10 +453,190 @@ NelsonLaws <- function(data, allsix = FALSE, chart = "i", xLabels = NULL) {
+    }
+    
+    if (identical(measurements, "") && !identical(variable, "")) {
+      ppPlot$dependOn(optionContainsValue = list(variables = variable))
+      data <- data.frame(process = dataForPlot[[variable]])
+      sixsigma_I <- qcc::qcc(data$process, type ='xbar.one', plot=FALSE)
+      xmr.raw.r <- matrix(cbind(data$process[1:length(data$process)-1], data$process[2:length(data$process)]), ncol = options$ncol)
+      sixsigma_R <- qcc::qcc(xmr.raw.r, type="R", plot = FALSE)
+    } else {
+      data <- as.vector((t(dataForPlot[measurements])))
+      sixsigma_I <- qcc::qcc(data, type ='xbar.one', plot=FALSE)
+      xmr.raw.r <- matrix(cbind(data[1:length(data)-1],data[2:length(data)]), ncol = 2)
+      sixsigma_R <- qcc::qcc(xmr.raw.r, type="R", plot = FALSE)
+    }
+    dataPlotI[[1]] <- c(unlist(dataPlotI[[1]]), unlist(sixsigma_I$statistics), min(sixsigma_I$limits), max(sixsigma_I$limits))  # all values and limits to calculate decimals and axes
+    if (i != 1) {
+      subgroupsI <- seq(max(dataPlotI[[i]]$subgroups) + 1, max(dataPlotI[[i]]$subgroups) + length(sixsigma_I$statistics))  # to keep counting across groups
+      subgroupsR <- seq(max(dataPlotR[[i]]$subgroups) + 1, max(dataPlotR[[i]]$subgroups) + length(sixsigma_R$statistics) + 1)
+      } else {
+      subgroupsI <- c(1:length(sixsigma_I$statistics))
+      subgroupsR <- seq_len(length(sixsigma_R$statistics) + 1)
+    }
+    dataPlotI[[i + 1]] <- list(subgroups = subgroupsI, process = sixsigma_I$statistics, center = sixsigma_I$center,
+                               UCL = max(sixsigma_I$limits), LCL = min(sixsigma_I$limits), sixsigma_I = sixsigma_I)
+    dataPlotR[[1]] <- c(unlist(dataPlotR[[1]]), unlist(sixsigma_R$statistics), max(sixsigma_R$limits, min(sixsigma_R$limits)))
+    dataPlotR[[i + 1]] <- list(subgroups = subgroupsR, movingRange = c(NA, sixsigma_R$statistics),
+                            center = sixsigma_R$center, UCL = max(sixsigma_R$limits), LCL = min(sixsigma_R$limits), sixsigma_R = sixsigma_R)
+  }
+  decimals1 <- max(.decimalplaces(unlist(dataPlotI$allValues)))
+  decimals2 <- max(.decimalplaces(dataPlotR$allValues))
+  yBreaks1 <- jaspGraphs::getPrettyAxisBreaks(dataPlotI$allValues)
+  yBreaks2 <- jaspGraphs::getPrettyAxisBreaks(dataPlotR$allValues)
+
+  # remove all values list, to allow looping over other lists
+  dataPlotI <- dataPlotI[-1]
+  dataPlotR <- dataPlotR[-1]
+  
+  plotMat <- matrix(list(), 2, nStages)
+
+  # Create plots
+  for (i in seq_len(nStages)) {
+    subgroups = dataPlotI[[i]]$subgroups
+    center <- dataPlotI[[i]]$center
+    UCL <- dataPlotI[[i]]$UCL
+    LCL <- dataPlotI[[i]]$LCL
+    if (options$manualTicks){
+      nxBreaks <- options$nTicks
+      xBreaks1 <- as.integer(c(jaspGraphs::getPrettyAxisBreaks(subgroups, n = nxBreaks)))
+    } else {
+      xBreaks1 <- as.integer(c(jaspGraphs::getPrettyAxisBreaks(subgroups)))
+    }
+    if (min(xBreaks1) == 0)  # never start counting at 0 on x axis
+      xBreaks1[1] <- 1
```

Comment:
```suggestion
    if (xBreaks1[1] == 0)  # never start counting at 0 on x axis
      xBreaks1[1] <- 1
```
since you're assuming that `xBreaks1[1] == min(xBreaks1)` anyway?

Reply:
Indeed